### PR TITLE
Add `ConnectionLivenessCheckTimeout` configuration

### PR DESCRIPTION
**Pull request by StephenCathcart**
This configuration defines the duration that the connection may be idle before needing to perform a liveness check on acquiring from the pool.

```golang
driver, err := neo4j.NewDriverWithContext("neo4j://localhost:7687", neo4j.BasicAuth("neo4j", "password", ""), func(config *neo4j.Config) {
    config.ConnectionLivenessCheckTimeout = 2 * time.Second
})
```

Check the API docs for more information.

### CODE REVIEW

**Comment by StephenCathcart** [comment]
[[go] Implement ADR 021: liveness check timeout](https://trello.com/c/9jTJ31i4/124-go-implement-adr-021-liveness-check-timeout)

File: neo4j/config.go [hunk 1]
```diff
@@ -77,6 +79,11 @@ func validateAndNormaliseConfig(config *Config) error {
 		config.ConnectionAcquisitionTimeout = -1
 	}
 
+	// Connection Liveness Check Timeout
+	if config.ConnectionLivenessCheckTimeout < 0 {
+		return &UsageError{Message: "Connection liveness check timeout cannot be smaller than 0"}
+	}
```
Comment (robsdedude) [inline neo4j/config.go:85]: Consider adding a unit test for this case (there already is a `TestValidateAndNormaliseConfig` that you can extend).

File: neo4j/internal/router/router_test.go [hunk 2]
```diff
@@ -56,7 +57,7 @@ func TestMultithreading(t *testing.T) {
 		n = n.Add(time.Duration(table.TimeToLive) * time.Second * 2)
 		return n
 	}
-	router := New("router", func() []string { return []string{} }, nil, pool, logger, "routerid", &timer)
+	router := New("router", func() []string { return []string{} }, nil, pool, math.MaxInt64, logger, "routerid", &timer)
```
Comment (robsdedude) [inline neo4j/internal/router/router_test.go]: Any reasons for/against using `DefaultConnectionLivenessCheckTimeout` here instead?

**Review by robsdedude — APPROVED** [review]
You might want to add a TestKit test skip in accordance with the decision documented in [the ADR update (internal link)](https://github.com/neo-technology/drivers-adr/pull/83). Something along the lines of [this](https://github.com/neo4j/neo4j-python-driver/commit/20126112dde1b82da0c441c0232ff756f0830b5b).

File: neo4j/config.go (hunk 1 above)
Comment (StephenCathcart) [reply neo4j/config.go:85]: Yep agree 👍

**Pull request merged**